lib/spanner/config/semantic_validator.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Spanner.Config.SemanticValidator do
+
+  alias Piper.Permissions.Ast
+  alias Piper.Permissions.Parser
+
+  @moduledoc """
+  Validates bundle config semantics.
+  """
+
+
+  @doc """
+  Accepts a config map and validates semantics.
+  """
+  @spec validate(Map.t) :: :ok | {:error, [{String.t, String.t}]}
+  def validate(config) do
+    verify_rules(config)
+  end
```

Comment:
The thinking was that we could add additional semantic validations as the need arose. We only validate rules now, but eventually we may have something like:
```
with :ok <- verify_rules(config),
        :ok <- verify_perms(config),
        :ok <- verify_other_things_that_need_to_be_verified(config) do
  :ok
end
```